compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -536,8 +536,8 @@ object ProtoTypes {
     def typedArg(arg: untpd.Tree, formal: Type)(using Context): Tree = {
       val wideFormal = formal.widenExpr
       val argCtx =
-        if wideFormal eq formal then ctx
-        else ctx.withNotNullInfos(ctx.notNullInfos.retractMutables)
+        if wideFormal eq formal then ctx.retractMode(Mode.InAnnotation)
+        else ctx.retractMode(Mode.InAnnotation).withNotNullInfos(ctx.notNullInfos.retractMutables)
```

Comment:
Can this flag be added to argCtx in https://github.com/scala/scala3/blob/4518ce5e8c402e5001eefd062f3bd6fd34627642/compiler/src/dotty/tools/dotc/typer/Applications.scala#L1063 instead ? That seems more generic. Also the reason for doing that should be documented (and the flag InAnnotation should document that it's turned off for arguments).

Reply:
> Can this flag be added to `argCtx`

No, that would remove the mode too early. `argCtx` is passed to `ApplyToUntyped` and is then used from the initializer of the parent class `TypedApply`, where we still need the mode to be set when typing an annotation.

So I think it's only in `ApplyToUntyped.typedArg` that we can remove the flag, which directly calls `ProtoTypes.typedArg`.

Reply:
I added documentation, tell me if that's clearer.

Reply:
Yes, thanks!